### PR TITLE
Sa150/text input

### PR DESCRIPTION
**Pull request by sanjayssj5**
# Description

Adding a common Input Text component for reuse

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

# Reference JIRA tickets : 

- SA-#150

# Checklist:

Put an "x" in the brackets to check the checkbox.

- [x ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in particularly hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [x] I have made corresponding changes to JIRA tickets
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have merged develop into my branch and resolved possible conflicts

### CODE REVIEW

**Comment by sanjayssj5** [comment]
Text input  usage was found in upload modal in home page and sidebar error components(used in knowledge graph and summary components). For the CSS part both of them are not similar so used them as it is for now.

File: src/index.tsx [hunk 1]
```diff
@@ -20,5 +20,5 @@ root.render(
 
 // If you want to start measuring performance in your app, pass a function
 // to log results (for example: reportWebVitals(console.log))
-// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+////// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
```
Comment (vlahovivan) [inline src/index.tsx]: where did this come from?

**Review by vlahovivan — CHANGES_REQUESTED** [review]
It works well, the only thing I've immediately noticed is that all of the InputText components have the same ID. That causes the suggestions (search history) to be the same for all instances of input. I would suggest adding a custom domID to the input in the upload modal component (something like `sa-upload-modal-input-text`) and doing the same thing for sidebar error (something like `sa-sidebar-error-input-text`).

**Comment by sanjayssj5** [comment]
Made the changes. The id must be unique now.

File: src/index.tsx (hunk 1 above)
Comment (sanjayssj5) [reply src/index.tsx]: not sure about this XD

File: src/components/InputText/InputText.tsx [hunk 2]
```diff
@@ -4,9 +4,10 @@ import cs from "classnames";
 import { InputTextProps } from "./InputText.types";
 
 const InputText: FC<InputTextProps> = ({
-  domID = "input-text",
-  dataTestId = "test-input-text",
   className,
+  domID = className?.concat("-input-text"),
```
Comment (vlahovivan) [inline src/components/InputText/InputText.tsx]: I don't think this is an ideal solution, for example if one element has a className "a b c" and the other one has className "c b a" - they will get different IDs even if you expect them to be the same.
I think you should leave this domID as it was, and change it only in InputText instances in UploadModal and SidebarError components

**Review by vlahovivan — APPROVED** [review]

**Pull request merged**